cpp/src/io/orc/orc.h

```diff
@@ -609,6 +643,123 @@ class metadata {
   mutable std::vector<std::string> column_names;
 };
 
+/**
+ * @brief `column_device_view` and additional, ORC specific, information on the column.
+ */
+struct orc_column_device_view {
+  column_device_view cudf_column;
+  thrust::optional<uint32_t> parent_index;
+};
+
+/**
+ * @brief Helper class that adds ORC-specific column info
+ */
+class orc_column_view {
+ public:
+  /**
+   * @brief Constructor that extracts out the string position + length pairs
+   * for building dictionaries for string columns
+   */
+  explicit orc_column_view(uint32_t index,
+                           std::optional<uint32_t> str_idx,
+                           int index_in_table,
+                           column_view const& col,
+                           const table_metadata* metadata);
+
+  auto is_string() const noexcept { return cudf_column.type().id() == type_id::STRING; }
+  void set_dict_stride(size_t stride) noexcept { dict_stride = stride; }
+  auto get_dict_stride() const noexcept { return dict_stride; }
```

Comment:
This is just a preference but I prefer to leave off the `get_` on accessor methods. Modifying functions should be verbs (aka `set_`...) and non modifying functions should be nouns.

Reply:
Agreed. This is old code that we didn't update the naming of. I'll change it.